visma/functions/constant.py

```diff
@@ -48,14 +48,31 @@ def __radd__(self, other):
         return self + other
 
     def __add__(self, other):
-        if self.isZero():   # if one of them is Empty, we can return the other one even it is Empty too because we need at least one to be returned.
+        if isinstance(other, Constant):
+            if other.power == self.power:
```

Comment:
Simplify the constants before doing any operations (use `calculate()`). I think this might result in `2^2 + 3^2 = 5^2`.

Reply:
done.